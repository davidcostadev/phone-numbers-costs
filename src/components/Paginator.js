import React from 'react';
import PropTypes from 'prop-types';

const Paginator = ({ onClickPage }) => (
  <div className="paginator">
    <button type="button" className="paginator__item paginator__item--disabled">Previews</button>
    <button type="button" id="paginator-1" className="paginator__item paginator__item--active" onClick={() => onClickPage(1)}>1</button>
    <button type="button" id="paginator-2" className="paginator__item" onClick={() => onClickPage(2)}>2</button>
    <button type="button" id="paginator-3" className="paginator__item" onClick={() => onClickPage(3)}>3</button>
    <button type="button" className="paginator__item">Next</button>
  </div>
);

Paginator.propTypes = {
  onClickPage: PropTypes.func.isRequired,
};

export default Paginator;
