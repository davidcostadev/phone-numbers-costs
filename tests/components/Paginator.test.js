import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Paginator from '../../src/components/Paginator';

configure({ adapter: new Adapter() });

describe('Paginator Component', () => {
  let onClickPage;

  beforeEach(() => {
    onClickPage = jest.fn();
  });

  describe('Render', () => {
    it('default', () => {
      const tree = renderer.create((
        <Paginator onClickPage={onClickPage} />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('click on onClickPage', () => {
      const wrapper = shallow(<Paginator onClickPage={onClickPage} />);

      wrapper.find('#paginator-1').simulate('click');
      expect(onClickPage).toBeCalledWith(1);

      wrapper.find('#paginator-2').simulate('click');
      expect(onClickPage).toBeCalledWith(2);

      wrapper.find('#paginator-3').simulate('click');
      expect(onClickPage).toBeCalledWith(3);
    });
  });
});
