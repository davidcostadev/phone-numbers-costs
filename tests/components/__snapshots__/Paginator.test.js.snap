// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component Render default 1`] = `
<div
  className="paginator"
>
  <button
    className="paginator__item paginator__item--disabled"
    type="button"
  >
    Previews
  </button>
  <button
    className="paginator__item paginator__item--active"
    id="paginator-1"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="paginator__item"
    id="paginator-2"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="paginator__item"
    id="paginator-3"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="paginator__item"
    type="button"
  >
    Next
  </button>
</div>
`;
